docs/source/tutorial/dags.rst

```diff
@@ -0,0 +1,288 @@
+***********************
+Directed Acyclic Graphs
+***********************
+
+This tutorial will explore using retworkx to work with directed acyclic graphs
+(also known as DAGs).
+
+Directed Graph
+==============
+
+As a primer, let's first review the broader concept of a directed graph (without the acyclic restriction).
+
+A directed graph is made up of a set of nodes connected by
+directed edges (often called arcs). Directed edges can only be followed in one direction,
+which is different from the edges found in undirected graphs, which can be followed
+in either direction. For example:
+
+.. jupyter-execute::
+
+    import retworkx as rx
+    from retworkx.visualization import mpl_draw
+
+    path_graph = rx.generators.directed_path_graph(5)
+    mpl_draw(path_graph)
+
+In this example, we've created a 5 node directed path graph. The directionality of
+each edge is denoted by the arrow head, which points to the target node. The node at
+the other end of the edge is called the source.
+
+Directed Acyclic Graphs
+=======================
+
+A directed acyclic graph is a directed graph which also doesn't contain
+any cycles. A cycle is a non-empty trail [#]_ in which the first and last nodes in
+the trail are the same. For example:
+
+.. jupyter-execute::
+
+   cycle_graph = rx.generators.directed_cycle_graph(5)
+   mpl_draw(cycle_graph)
+
+is **not** acyclic. While the earlier path graph is acyclic.
+
+In retworkx you can create a new :class:`.PyDiGraph` object that enforces
+that no cycle is added by using the ``check_cycle`` constructor property
+
+.. jupyter-execute::
+
+    dag = rx.PyDiGraph(check_cycle=True)
+
+or after a :class:`~retworkx.PyDiGraph` object is created with the
+:attr:`~retworkx.PyDiGraph.check_cycle` attribute:
+
+.. jupyter-execute::
+
+    dag.check_cycle = True
+
+You can also check the status of cycle checking by inspecting the
+:attr:`~retworkx.PyDiGraph.check_cycle` attribute:
+
+.. jupyter-execute::
+
+    print(dag.check_cycle)
+
+With ``check_cycle`` set to ``True``, methods that add edges
+to the graph (e.g. :meth:`.PyDiGraph.add_edge`) will raise an error
+if a cycle would be introduced. This checking
+does come with a noticeable (and potentially large) runtime overhead so it's
+best to only use it if it's strictly necessary. You can also avoid this
+overhead by using :meth:`.PyDiGraph.add_parent` and
+:meth:`.PyDiGraph.add_child` as both add a new node and edge simultaneously and
+neither can introduce a cycle.
+
+.. [#] A trail is a sequence of nodes and edges on the graph where each edge
+   is distinct in the walk.
```

Comment:
Can you put the footnote content inside a `.. note::` as well? It looks awkward and hard to differentiate from the text in the current format. I know it is a limitation of the documentation theme but I think asking Sphinx to explicitly use note would make it look better

Reply:
Oops I missed this comment. Given that I'd like to get the release out today and CI is almost done I think we can fix this in a follow-up easily enough post release. So I'll leave it for then.